### ISSUE TITLE
Player Weapon should always be rendered

### ISSUE THREAD

**Issue by rvanner** [TheGameCreators/GameGuruRepo#1839]
Here's an example where the player weapon pokes into a dead enemy. It should be forced to always be drawn last.

![image](https://user-images.githubusercontent.com/1503593/151048760-533a82d8-cc54-48bd-be2b-6a238a98da78.png)

**Comment by UltraVox001** [comment]
It's the same on the walls and the trees...

**Comment by ArgentArts** [comment]
Isn't that a **collision** issue? The gun is penetrating the enemy or wall or tree and so that part is not being drawn.

**Comment by UltraVox001** [comment]
Yes, it's possible. I've already tried to fix this by adding all sorts of collisions to the weapon, but considering Max's current collision issues, it didn't change anything.

@rvanner : can you update us on this, please ?
Will Max's Item Library collision issues be fixed soon ? It's a **game breaker**.

I invite you to watch my short video :
[https://www.youtube.com/watch?v=Tx_c1sJna7Q](https://www.youtube.com/watch?v=Tx_c1sJna7Q)
And that of Pirate Mike :
[https://www.youtube.com/watch?v=WlB2c7_Maeo](https://www.youtube.com/watch?v=WlB2c7_Maeo)

**Comment by rvanner** [comment]
Weapons should be rendered differently. You just draw them last so that they never poke into anything. Lee will sort it in Feb.
For the other asset collision issues, please post as a separate issue and we'll deal with them in Feb.

**Comment by UltraVox001** [comment]
@rvanner 
It's already done : [1796](https://github.com/TheGameCreators/GameGuruRepo/issues/1796)
I inform you that this problem prevents to build demo maps for TGC.
This is a very serious problem that should be fixed as a Top Priority.

**Comment by rvanner** [comment]
@LeeBamberTGC Can you take a quick look at this collision issue today?

**Comment by LeeBamberTGC** [comment]
@rvanner Looks like a bug when the collision type is changed in the level editor from its default. Please do report any objects that have incorrect collision types by default, before they are manually changed. Thank you.

**Comment by UltraVox001** [comment]
@LeeBamberTGC and @rvanner 
The problem is maybe elsewhere.

On an empty map, you won't find any collision problems. But as soon as there are other objects on the map, the problem occurs. Worse still : the problem is "transferred" to other objects that were not affected by this collision problem.

Test 1 with object : Rusty Small Industrial Shelves
It appears with Polygonal collisions by default, but I can get through it in the TestLevel.
[Test 1](https://www.youtube.com/watch?v=Tx_c1sJna7Q)

Test 2 with object : Rusty Small Industrial Shelves
It appears with Hull Decomp collisions, but I can't get through it in the TestLevel.
![HD](https://user-images.githubusercontent.com/96718175/151138346-b9d8687f-1708-48a4-82db-4e55dde081f5.jpg)

There are objects that appear to "collision conflict" with other objects previously dropped on the map. And Max can't tell who wants what. It assigns any type of collision, to any object.

**Comment by ArgentArts** [comment]
Out of curiosity, try this for a test using "Test 1" - raise the Rusty Small Industrial Shelves up off the ground so it's not touching the ground or anything and set "IsImmobile" and your collision setting of choice, then run the game. Can you still walk through it?

What I want to see here is, if the entity is off the ground and collision is being kept, then perhaps this is a clue that the auto-nav mesh was messing with it and changing the collision mesh.

**Comment by UltraVox001** [comment]
In Test 1, with the same object, located higher and the "IsImmobile" box is checked.
It appears with Hull Decomp collisions by default, and I can walk through it.
[Hull Decomp](https://i.gyazo.com/ca3a480dfca2923af3f853b726ec5d1b.jpg)

After the TestLevel, if I click again on the object, it is in No Collision :
[No Collision](https://i.gyazo.com/a6957db40ac3beed89238bbadcf5a122.jpg)

So, I change the collisions to Hull Decomp, but still get through it.
I've tried every other collision mode, with "isImmobile" checked, but I still get through it.

Looks like Max is attributing collisions of another object to this one. Or, Max can't read or write data, and can't understand it when asked.

**Comment by ArgentArts** [comment]
Interesting. Thanks for trying that. At least we can guess that it's not the intersecting of the entity with the nav mesh that's causing this. Hopefully Lee will sort this out soon. I'd be interested to find out what is causing the issue.

**Comment by UltraVox001** [comment]
Except that the object is first placed on the ground, and then I raise it. I will try again by dropping the object in the air (without touching the ground). Well, if Max allows me. :)

**Comment by UltraVox001** [comment]
This is the same phenomenon as the previous one.
First in Hull Decomp and I walk through.
Then I click on it and it shows No Collision...

**Comment by ArgentArts** [comment]
Well, I was trying to think of something to try. ;)

**Comment by UltraVox001** [comment]
We are here for that ! :)
It's mysterious. There is a conflict somewhere...
And it ends like this : [Video](https://i.gyazo.com/a6f0f11966f5fe8b509d41d649546730.mp4)
It refuses to change the collisions and we are stuck with an unusable map :(

**Comment by ArgentArts** [comment]
There are several things that are currently making MAX unusable - this collision issue, the lighting issue (where lighting suddenly gets brighter/dimmer depending on distance, etc.), strange snapping issues when using the editor, etc.

**Comment by UltraVox001** [comment]
For the lighting problem, it's the same principle as the tests I just did with collisions : no problem with lights on an empty map. Even at long distances, everything is very well lit. As soon as there are objects here and there, the problem arises. You move 5 or 10 meters away from a Light, and it disappears.

**Comment by ArgentArts** [comment]
I've had other issues, though. Sometimes the lighting changes where it get brighter the further out a camera goes or an object that was casting a shadow (by blocking the light) stops casting a shadow, etc.

**Comment by UltraVox001** [comment]
I removed Auto Exposure from Post Processing to try to minimize this issue.

**Comment by ArgentArts** [comment]
I don't use auto exposure. ;)

**Comment by UltraVox001** [comment]
@rvanner  and @LeeBamberTGC 
Given the existence of this major bug with collisions, as well as other bugs (Crash, Light...) that prevent some developers from offering you demo maps, I suggest you reconsider the deadline for sending you decent demo maps. Unless Lee knows how to fix this bug very quickly. But the date of January 31 is much too close (5 days) to send you a functional map. Most of the users of the public version of Max EA have been stuck for weeks because of this. We even talk about it on the Gameguru forum, but the info is only coming to you now.

**Comment by LeeBamberTGC** [comment]
@UltraVox001 It is certainly a bug, and will be fixed soon. Do not worry about demo maps that allow the player to walk THROUGH things, that will be instantly fixed when the collision bug is fixed. We will judge demo levels on the assumption all collision issues are fixed. We certainly will not penalise based on a bug we created :)

**Comment by ArgentArts** [comment]
Yes, but it is difficult for the end-user to test their own levels when they can walk through their own walls and entities. It is difficult to test the PLAYABILITY of their own levels when their collision is not working. It's not just about you judging their levels, it's also about them being able to TEST their own levels.

**Comment by ArgentArts** [comment]
I had a level I was working on that had the opposite problem that others here were experiencing. Instead of collision being turned off, the entity was losing it's polygonal collision and getting box collision instead. This was an intricate door frame and, as a result, I could not walk through the door when it opened as the box collision of the frame was preventing me from passing. This was working previously, but something changed at one point and it stopped working for some unknown reason. As a result, I could no longer test this level. Problems like this make it difficult, if not impossible, for the end-user to fully test THEIR OWN levels. So, this is not just about you judging their levels. ;)

**Comment by UltraVox001** [comment]
In reality the problem is the same. I gave examples of missing collisions, but I have many other examples of Polygons objects turned into Box collisions. The only levels that can work "properly" are those outdoors, without any buildings or 3D objects of any kind. Like zombies in a forest. But even there, we have problems with Zones and Scripts...

In other words : we can't test anything.

**Comment by LeeBamberTGC** [comment]
Email me at lee@thegamecreators.com and I will add you the internal testers list, and let anyone else know who you see are having the same issues they can do likewise. Essentially the requirement is that you test and report prior to the public release, but I can always remove any user who wishes after the competition ends.  We can then get the collision fix to you today (if I can easily reproduce with stock art and the current build) :)

**Comment by LeeBamberTGC** [comment]
P.S. I will need your TGC forum name you used to get MAX!

**Comment by UltraVox001** [comment]
My nickname on the forum is UltraVox.
I send you an email.

**Comment by UltraVox001** [comment]
@LeeBamberTGC and @rvanner 
I think I found the origin of this momentary bug : Sometimes, Max allows himself to copy/paste all the 3D elements of the map, under the map (under the ground). There are therefore several instances of the same 3D objects, and when Max is asked to modify the collisions of one of them, he cannot understand this request.

I found the origin of the bug while trying to solve another one, located on the same map! In reality, it was the same bug that produced these two problems :)
[Bug Found](https://github.com/TheGameCreators/GameGuruRepo/issues/1846)

**Comment by rvanner** [comment]
@UltraVox001 Are you able to make this duplicating bug happen every time or is it just random?

**Comment by UltraVox001** [comment]
@rvanner 
I built 16 maps on which this bug occurred. I had to zoom out the map to the limit area delimited in yellow, then I made a selection with the mouse and Shift. I found a multitude of diverse and varied objects, hidden under the map. However, I would never have asked Max to copy/paste a house containing more than 600 3D objects, under the map.

Like on this screenshot. Some objects is selected, in the distance, under the map, while my only construction is a house at the bottom of the screen.
![CopyUnderTheMap](https://user-images.githubusercontent.com/96718175/151334361-28e8a81a-7870-4f00-814b-b401b8e772b5.jpg)

Sometimes the simple act of selecting an object generates a copy/paste. And the fact of selecting a house which contains a lot of things, makes them appear too, in yellow (thus, selected with the house). This is a problem.

There may be a bug, or an improvement to be made, in the Shift+Left Clik command to copy an object. This command is too "accessible" and causes problems. If you go a little too fast in your object selection, you risk copying/pasting a whole set of objects that will hide under the map! It's the same bug as the one you found a few days ago, with the Characters that you move with the mouse, but which go away : [Your video](https://www.youtube.com/watch?v =QOyi16qTC3Y)
It's the same as here, but with an unrequested copy/paste.

Sometimes they are copied close to the spawn (StartPlayer), under the map. Sometimes at very long distances, but almost always below ground.

**Comment by UltraVox001** [comment]
Maybe a tick option in the general settings would help.

- Ask permission to copy/paste an object

If the box is checked, the user must Confirm that he wants to copy an object. (for each copy/paste)

If unchecked, then copy/paste occurs, as it does now.

**Comment by UltraVox001** [comment]
If it helps Lee, I have a map in which 70% of 3D objects no longer have any collisions.
It's impossible to give them collisions from the Max interface.

**Comment by UltraVox001** [comment]
@LeeBamberTGC 
Did you receive my map? (Map16.fpm) I sent it to you by email.
You will find dozens of objects without the slightest collision. Starting with the Fence around the house.

I'm going to reload the little map you mention and in which Rusty Small Industrial Shelves refuses its collisions.

**Comment by UltraVox001** [comment]
@LeeBamberTGC and @rvanner 
Be sure that I am not the only one to whom this problem occurs :
![userprob1](https://user-images.githubusercontent.com/96718175/151394307-92c6d690-e11d-42a7-b02a-9dbe782f57e5.png)
![userprob2](https://user-images.githubusercontent.com/96718175/151394316-7489eba2-5f76-4205-be0f-05eb133c2eea.png)
These screens refer to a Thread on the [GG Forum](https://forum.game-guru.com/thread/223291)

**Comment by LeeBamberTGC** [comment]
@UltraVox001 This has been fixed with the latest update, please check it out and let me know.

**Comment by LeeBamberTGC** [comment]
P.S> That is to say the collision bug has been fixed, but the original issue reported is still at large. We always recommend issues be separated and avoid cramming a second bug report into the first bug report, especially if they are not related. It just confuses everyone :)

**Comment by UltraVox001** [comment]
@LeeBamberTGC 

You are right. Receive my apologies. I took advantage of a Collisions Post to encompass another collision issue. The problem is solved for me and for others, but Not for Pirate Myke : [https://forum.game-guru.com/thread/223291#msg2646314](https://forum.game-guru.com/thread/223291#msg2646314)

In the future I will no longer write on this Post.
Thank you.
